### PR TITLE
Partipants Responses.

### LINKED ISSUE #5: Delete Poll
Goal: User should be able to delete a poll. A displayed poll should only have the option to delete if the correct user is signed in.

### LINKED ISSUE #17: Landing Page
Change '/' page to be landing page.

### PR DESCRIPTION
**Pull request by jduong09**
## Goal
- Allow user to visit shared poll link in order to add a response to poll.
- Allow admin to visit separate page to visualize results and make updates to poll.

### Future Additions
- Allow admin to add deadlines to when responses can be submitted.

### CODE REVIEW

**Review by ziggysauce — COMMENTED** [review]
hmm, this doesn't seem to be working for me. can you add more detailed notes/instructions on how to test your new features? additional images or videos would also help.

**Comment by jduong09** [comment]
Sorry there was no clear instruction on how to test the branch. I updated the branch. Should be clear now.
- To submit a response to the poll, you can use the 'Submit Response' button.
- To take a look at admin detail page, you can use the 'Admin' button. 
These buttons will be removed, and a landing page will replace all the polls that are displayed currently at the '/' route.

**Comment by jduong09** [comment]
## Updates
- Added delete poll button, which is located from the admin button on the landing page.

File: root/backend/server/api/routes/poll.js [hunk 1]
```diff
@@ -78,4 +80,39 @@ router.get('/:pollId/pollInfo', async (req, res) => {
...
+      return data.responses;
+    });
+  
+    let newResponseObj = {};
+
+    if (responsesData) {
+      newResponseObj = {
+        ...responsesData,
+      }
+    }
```
Comment (ziggysauce) [inline root/backend/server/api/routes/poll.js:98]: if you're not adding anything else to this, no need to use the spread operator here

```suggestion
    if (responsesData) {
      newResponseObj = responsesData;
    }
```

File: root/backend/server/api/routes/poll.js [hunk 2]
```diff
@@ -78,4 +80,39 @@ router.get('/:pollId/pollInfo', async (req, res) => {
...
+    choices.forEach((timestamp) => {
+      if (newResponseObj[timestamp]) {
+        newResponseObj[timestamp][userUuid] = name;
+      } else {
+        newResponseObj[timestamp] = {
+          [userUuid]: name
+        }
+      }
+    });
+    console.log(newResponseObj);
```
Comment (ziggysauce) [inline root/backend/server/api/routes/poll.js]: remove console log when done testing

File: root/backend/server/server.js [hunk 3]
```diff
@@ -26,6 +27,16 @@ db.on('error', console.error.bind(console, 'mongo connection error'));
 
 app.use(router);
 
+app.delete('/admin/:pollUuid', async (req, res, next) => {
+  try {
+    await Poll.findByIdAndDelete({ _id: req.params.pollUuid }).then(data => data);
```
Comment (ziggysauce) [inline root/backend/server/server.js]: what is the purpose of the `.then()` usage here?

File: root/client/src/components/poll/availabilitiesTable.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { React, useEffect, useState } from 'react';
+import { calculateTimeFrame, getDayOfTheWeek } from '../../util/tablehelpers';
+
+export const AvailabilitiesList = ({ availabilities, responses, duration, handleUserChoice }) => {
+  const [listItems, setListItems] = useState([]);
+  console.log(availabilities);
```
Comment (ziggysauce) [inline root/client/src/components/poll/availabilitiesTable.js]: remove console log when done testing

File: root/client/src/components/poll/pollDetailAdmin.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  const handleDelete = async (e) => {
+    e.preventDefault();
+    const response = await fetch(`/admin/${pollUuid}`, { method: 'DELETE' });
+
+    const data = response.json();
+
+    if (response.ok) {
+      console.log(data);
```
Comment (ziggysauce) [inline root/client/src/components/poll/pollDetailAdmin.js]: remove console log when done testing

**Review by ziggysauce — COMMENTED** [review]
ssiiiiick this appears to be working well.


i couldn't tell from your notes exactly which buttons are going away, but this "update" button from the admin page doesn't seem to do anything. When I click on it it just takes me to a blank page.

Screen Shot 2023-11-28 at 4 13 41 PM


otherwise this seems to be working all as intended. nice work

**Comment by ziggysauce** [comment]
this PR resolves https://github.com/jduong09/doodle/issues/17 too i believe

**Review by ziggysauce — CHANGES_REQUESTED** [review]
Looking good!! everything worked as expected. the only _weird_ thing i'm seeing is that the responses here are showing 5am - 6am on friday, but the poll doesn't actually have that. so something funky is going on with the time display

Screen Shot 2023-11-30 at 6 34 28 PM
Screen Shot 2023-11-30 at 6 34 46 PM

File: root/backend/server/server.js (hunk 3 above)
Comment (jduong09) [reply root/backend/server/server.js]: No purpose

File: root/backend/server/api/routes/poll.js (hunk 1 above)
Comment (jduong09) [reply root/backend/server/api/routes/poll.js:98]: Gotcha, thanks

**Review by ziggysauce — CHANGES_REQUESTED** [review]
So something weird is happening now, when I submit a response as personA, then submit as personB, the results for personA disappear. 

[Screen Recording 2023-12-07 at 12.25.52 PM.zip](https://github.com/jduong09/doodle/files/13603139/Screen.Recording.2023-12-07.at.12.25.52.PM.zip)

**Comment by jduong09** [comment]
> So something weird is happening now, when I submit a response as personA, then submit as personB, the results for personA disappear.
> 
> [Screen Recording 2023-12-07 at 12.25.52 PM.zip](https://github.com/jduong09/doodle/files/13603139/Screen.Recording.2023-12-07.at.12.25.52.PM.zip)

- Issue resolved. Responses were being deleted in the backend.

**Pull request merged**